File: business-easy-ui/src/app/core/services/unit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { UnitData } from '../models/unit-data.model';

@Injectable({providedIn: 'root'})
export class UnitService {
  unitUrl = "api/unit/all";
  public unitData: UnitData;
  constructor(private http: HttpClient) {
    http.get(this.unitUrl).subscribe(
      (data: UnitData) => {
        console.log(data);
        this.unitData = data;
      }
    )
  }

}

File: business-easy-core/src/main/java/com/businesseasy/core/services/unit/UnitService.java
package com.businesseasy.core.services.unit;

import com.businesseasy.core.common.model.UnitData;

import java.math.BigDecimal;

public interface UnitService {
    UnitData getAllUnitData();

    BigDecimal convert(Long from, Long to, BigDecimal amount);
}
